app/src/test/java/org/dhis2/usecases/datasets/dataSetInitial/DataSetInitialPresenterTest.kt

```diff
@@ -0,0 +1,141 @@
+import org.dhis2.usescases.datasets.datasetInitial.DateRangeInputPeriodModel
+import org.hisp.dhis.android.core.category.CategoryOption
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+import org.hisp.dhis.android.core.period.PeriodType
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import java.util.Date
+
+class DataSetInitialPresenterTest {
+
+    private lateinit var presenter: DataSetInitialPresenter
+
+    private val view: DataSetInitialView = mock()
+    private val repository: DataSetInitialRepository = mock()
+    private val scheduler = TrampolineSchedulerProvider()
+
+    @Before
+    fun setUp() {
+        presenter = DataSetInitialPresenter(view, repository, scheduler)
+    }
+
+    @Test
+    fun `Should setOrgUnit and data`() {
+
+        val orgUnits = listOf(OrganisationUnit.builder().uid("orgUnitUid").build())
+        val dataSet = DataSetInitialModel.create(
+            "name",
+            "desc",
+            "catComboUid",
+            "catCombo",
+            PeriodType.Daily,
+            mutableListOf(),
+            0
+            )
+        whenever(repository.orgUnits()) doReturn Observable.just(orgUnits)
+        whenever(repository.dataSet()) doReturn Observable.just(dataSet)
+
+        presenter.init()
+
+        verify(view).setOrgUnit(orgUnits[0])
+        verify(view).setData(dataSet)
+    }
+
+    @Test
+    fun `Should go back when back button is clicked`(){
+        presenter.onBackClick()
+
+        verify(view).back()
+    }
+
+    @Test
+    fun `Should show orgUnitDialog when field is clicked`() {
+        val orgUnits = listOf<OrganisationUnit>()
+
+        presenter.onOrgUnitSelectorClick()
+
+        verify(view).showOrgUnitDialog(orgUnits)
+    }
+
+    @Test
+    fun `Should show periodSelector when field is clicked`() {
+        val periodType = PeriodType.Monthly
+        val periods = listOf(DateRangeInputPeriodModel.create("dataSet", "period", Date(), Date(), Date(), Date()))
+
+        whenever(repository.dataInputPeriod) doReturn Flowable.just(periods)
+
+        presenter.onReportPeriodClick(PeriodType.Monthly)
+
+        verify(view).showPeriodSelector(periodType, periods, 0)
+    }
+
+    @Test
+    fun `Should show catOptionSelector when field is clicked`() {
+        val catOptionUid = "catOptionUid"
+        val catOptions = listOf(CategoryOption.builder().uid(catOptionUid).build())
+
+        whenever(repository.catCombo(catOptionUid)) doReturn Observable.just(catOptions)
+
+        presenter.onCatOptionClick(catOptionUid)
+
+        verify(view).showCatComboSelector(catOptionUid, catOptions)
+    }
+
+    @Test
+    fun `Should navigate to dataSetTableActivity when actionbutton is clicked`() {
+        val catCombo = "catComboUid"
+        val catOptionCombo = "catOptionCombo"
+        val periodId = "periodId"
+        val periodType = PeriodType.Monthly
+
+        `Should setOrgUnit and data`()
```

Comment:
you don't need to execute the other unit test. You can just call the whenevers :+1: